### ISSUE TITLE
Calling run_javascript within on_connect results in IndexError

### ISSUE THREAD

**Issue by hroemer** [zauberzeug/nicegui#219]
Calling `run_javascript()` inside an `on_connect` handler results in IndexError.

```
Task raised an exception
Traceback (most recent call last):
  File "/home/user/.cache/pypoetry/virtualenvs/ng_samples-iEghew-py3.10/lib/python3.10/site-packages/nicegui/task_logger.py", line 50, in _handle_task_result
    task.result()
  File "/home/user/Projects/ng_samples/custom_fapi/ui_samples.py", line 52, in countdown
    await execute_js()
  File "/home/user/Projects/ng_samples/custom_fapi/ui_samples.py", line 57, in execute_js
    result.text = await ui.run_javascript(f'document.getElementById("{label.id}").innerHTML;')
  File "/home/user/.cache/pypoetry/virtualenvs/ng_samples-iEghew-py3.10/lib/python3.10/site-packages/nicegui/functions/javascript.py", line 8, in run_javascript
    client = globals.get_client()
  File "/home/user/.cache/pypoetry/virtualenvs/ng_samples-iEghew-py3.10/lib/python3.10/site-packages/nicegui/globals.py", line 76, in get_client
    return get_slot().parent.client
  File "/home/user/.cache/pypoetry/virtualenvs/ng_samples-iEghew-py3.10/lib/python3.10/site-packages/nicegui/globals.py", line 72, in get_slot
    return get_slot_stack()[-1]
IndexError: list index out of range
```

Triggering it manually e.g. via a button works fine:

```python3
import asyncio
from datetime import datetime

from fastapi import FastAPI
from nicegui import ui


def create_clean(app: FastAPI, path: str = 'clean', title: str = 'clean'):
    @ui.page(path=path, dark=True, title=title)
    def page():
        with ui.row():
            ui.label(f'Hello, FastAPI! @{path}')
        with ui.row():
            ui.label(f'{title} page')

    ui.run_with(app=app)


def create_layout(app: FastAPI, path: str = 'clean', title: str = 'clean'):
    @ui.page(path=path, dark=True, title=title)
    async def page_layout():
        ui.label('CONTENT')
        with ui.header():
            ui.label('HEADER')
        with ui.left_drawer(top_corner=True, bottom_corner=True):
            ui.label('LEFT DRAWER')
        with ui.right_drawer(top_corner=False, bottom_corner=False):
            ui.label('RIGHT DRAWER')
        with ui.footer():
            ui.label('FOOTER')

    ui.run_with(app=app)


def create_connect(app: FastAPI, path: str = 'clean', title: str = 'clean'):
    label = None
    result = None

    @ui.page(path=path, dark=True, title=title)
    async def page_layout():
        nonlocal label, result
        ui.label(text=f'{datetime.now().isoformat()}: page loaded')
        label = ui.label(text=f'{datetime.now().isoformat()}: ')
        result = ui.label(text=f'{datetime.now().isoformat()}: js result')
        ui.button(text='trigger manually', on_click=execute_js)
        ui.on_connect(countdown)

    async def countdown():
        nonlocal label, result
        for i in [5, 4, 3, 2, 1, 0]:
            label.text = f'{datetime.now().isoformat()}: countdown {i}' if i else f'{datetime.now().isoformat()}: Take-off!'
            await execute_js()
            await asyncio.sleep(1)

    async def execute_js():
        try:
            result.text = await ui.run_javascript(f'document.getElementById("{label.id}").innerHTML;')
        except IndexError as ie:
            result.text = f'{datetime.now().isoformat()}: {ie}'

    ui.run_with(app=app)

app = FastAPI()
layout = create_layout(app=app, path='/connect', title='connect')
```

Is there some kind of race condition?

**Comment by falkoschindler** [comment]
Thanks for reporting this issue!

I boiled it down to the following example:

```python
from nicegui import ui

@ui.page(path='/')
def page():
    async def run_js():
        result.text = await ui.run_javascript('Date()')
    result = ui.label()
    ui.on_connect(run_js)
    ui.button(text='Run JS', on_click=run_js)

ui.run()
```

**Comment by falkoschindler** [comment]
This should be fixed in [acf1f8d](https://github.com/zauberzeug/nicegui/commit/acf1f8db396e34ad602416a6f0efe9e7dfa22be9).
